sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -528,13 +532,35 @@ abstract class GpuExplodeBase extends GpuUnevaluableUnaryExpression with GpuGene
   /**
    * A function that will do the explode or position explode
    */
-  private[this] def explodeFun(inputTable: Table, genOffset: Int, outer: Boolean): Table = {
+  private[this] def explodeFun(
+      inputTable: Table,
+      genOffset: Int,
+      outer: Boolean,
+      fixUpOffset: Long): Table = {
     if (position) {
-      if (outer) {
+      val toFixUp = if (outer) {
         inputTable.explodeOuterPosition(genOffset)
       } else {
         inputTable.explodePosition(genOffset)
       }
+      // the position column is at genOffset and the exploded column at genOffset + 1
+      val posColToFix = toFixUp.getColumn(genOffset)
```

Comment:
nit toFixUp needs to either be under withResource, or at least a closeOnException.